pydantic/fields.py

```diff
@@ -76,8 +77,12 @@ class FieldInfo(Representation):
         'extra',
     )
 
-    def __init__(self, default: Any, **kwargs: Any) -> None:
+    def __init__(self, default: Any = Undefined, *, default_factory: Any = None, **kwargs: Any) -> None:
+        if default is not Undefined and default_factory is not None:
```

Comment:
perhaps better to move this check into `Field`m then you can use `pop()` as is done elsewhere.